regression/iterate_data.yaml

```diff
@@ -6,25 +6,41 @@ command: |-
   pip uninstall -y mosaicml-streaming
   cd streaming
   pip install -e '.[dev]'
-  composer -n 1 regression/synthetic_dataset.py --create --compression gz
-  composer -n 2 regression/iterate_data.py
-  composer -n 1 regression/synthetic_dataset.py --delete
-  composer -n 1 regression/synthetic_dataset.py --create --hashes sha1 xxh128
-  composer -n 2 regression/iterate_data.py
-  composer -n 4 regression/iterate_data.py --local
-  composer -n 8 regression/iterate_data.py --download_retry 10
-  composer -n 8 regression/iterate_data.py --download_timeout 120
-  composer -n 8 regression/iterate_data.py --validate_hash sha1
-  composer -n 8 regression/iterate_data.py --validate_hash xxh128
-  composer -n 8 regression/iterate_data.py --keep_zip
-  composer -n 8 regression/iterate_data.py --predownload 1000
-  composer -n 8 regression/iterate_data.py --cache_limit 64mb
-  composer -n 8 regression/iterate_data.py --num_canonical_nodes 16
-  composer -n 8 regression/iterate_data.py --batch_size 1000
-  composer -n 8 regression/iterate_data.py --shuffle --shuffle_algo py1b --shuffle_seed 12 --shuffle_block_size 10000
-  composer -n 1 regression/synthetic_dataset.py --delete
-
-image: mosaicml/composer:0.15.0
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/ --num_samples 10_000 --size_limit 10240
+  composer -n 2 regression/iterate_data.py --remote /tmp/streaming_dataset/ --batch_size 16
+  composer -n 2 regression/iterate_data.py --remote /tmp/streaming_dataset/ --local /tmp/local_dataset/ --batch_size 32
+  composer -n 4 regression/iterate_data.py --local /tmp/streaming_dataset/
+  python regression/synthetic_dataset.py --delete --out /tmp/streaming_dataset/
+  python regression/synthetic_dataset.py --delete --out /tmp/local_dataset/
```

Comment:
This seems like a hacky way to delete the folder created by iterate_data.py. synthetic_data.py should only manage the folders it created, so maybe we have a separate cleanup file that deletes directories from iterate_data.py and synthetic_data.py?

Reply:
Sure. Another way is to just run `rm -rf /tmp/local_dataset/` !!!